### PR TITLE
[MIRROR] Internals Bugfixes & Internals-Compatible Helmets [MDB IGNORE]

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/70750
--------------------
## About The Pull Request
This PR contains a modest refactor, 1 major and 2 minor feature additions, and bugfixes, for several portions of the code pertinent to internals / mob-equipped air tanks.

The major additional feature of this PR adds internals support to many sealed helmets by adding a new `HEADINTERNALS` flag to them. The minor additions are AltClick on internals tanks, and an "externals" system for @ Fikou

#### Internals-Compatible Helmets

There are several careful considerations behind the changes to helmets, such as if a helmet can theoretically connect to an air tank based on its physical design. Specifically, I added internals support to helmets which at least fully enclose the user's head, and which satisfy at least one of my following criteria:

- _Criterion A_. The helmet must be hermetically sealed, i.e. it must at least have the `STOPSPRESSUREDAMAGE` flag. This includes all spacesuit helmets, space-worthy MOD suit helmets, the changeling's fleshmass, the hardened cult hoodie, and the atmospherics helmet.
- _Criterion B_. The helmet may not be fully sealed but must block significant airflow, i.e. it must have the `BLOCK_GAS_SMOKE_EFFECT` and `PEPPERPROOF` flags. This includes the biohood.

Additionally, in order to get MOD suit helmets to toggle internals as expected, I hooked into the `mod/control/proc/seal_part` proc to check internals status.

#### External Internals Tanks

I was also asked to add an "externals" system to carbons which allows them to breathe from an air tank which they have not equipped. One of the later first uses for such an externals hook will perhaps be a MOD suit module.

#### AltClick on Internals Tanks

I have added the AltClick proc to internals tanks `/obj/item/tank/internals` which allows humans to toggle their internals quickly.

### BugFixes & Internals Refactor

My original intention was only to add internals compatibility to helmets, however I encountered several bugs in the related code which this PR includes fixes for. The bug fixes included in this PR correct the following issues:

1. When a human adjusts any mask, its `wear_mask_update` function and `update_action_buttons_icon` procs are called in inappropriate conditions, i.e. when the mask was not equipped by that mob.
2. The Alt+click functionality of masks isn't working, due to a typo (stray `!`) in the `mask/breath/AltClick` proc.
3. When opening internals, any worn mask is adjusted over the face regardless of whether or not it's compatible with internals, due to `tank/toggle_internals` not checking for the `MASKINTERNALS` flag first.
4. When a human gives their open internals tank to another mob via the stripping menu, the tank can be opened for both mobs at once for a limited but noticeable period of time.

To fix the aforementioned bugs and easily add internals support to helmets, I quickly realized that I had to address a code health issue first. More concisely, I realized that I had the opportunity to improve relevant critical sections of code which interoperated with the code I was working on. To that end, I added several ubiquitous utility procs which allowed me to make the code a lot more DRY, eliminate repetitive and unreliable conditionals, separate some concerns between `tank` and `mob`, and ultimately better centralize the logic for toggling internals.

I wound up porting internals functionality entirely out of `/obj/item/tank` and into `/mob/living/carbon`. Several utility procs for working with internals were also added to `mob/living/carbon` and `mob/living/carbon/human`, allowing carbons to manage their own internals via `open_internals(tank)` and `can_breathe_internals()`etc.

To fix bugs which allowed multiple mobs to breathe from the same tank, the following procs were also added to `obj/item/tank`:

- `equipped(mob/living/user, slot, initial)` - Closes the open air tank if it was equipped by a mob other than the current user.
- `dropped(mob/living/user, silent)` - Closes the open air tank if it got dropped by it's current user.
- `after_internals_opened(mob/living/carbon/human/human_target)` - Called by carbons after they connect the tank to their breathing apparatus.
- `after_internals_closed(mob/living/carbon/human/human_target)` - Called by carbons after they disconnect the tank from their breathing apparatus.

## Why It's Good For The Game
This PR introduces a feature which allows several sealed helmets, such as spacesuit helmets, to be used as breathing apparatuses. Where appropriate, this PR adds internals compatibility to spacesuit helmets, space-worthy MOD suit helmets, the biohood, changeling fleshmass, hardened cult hoodie, and atmos helmet. I think most people would agree that in-general a space suit _should be_ capable of handling the connection to an internal air supply, and that they can easily do so without an extra mask attached directly to the face. This PR also includes a refactor to internals code in order to improve overall code health and fix several bugs.

The most noticeable bug fixed was that the Alt-click adjustment function of breath/gas masks was non-functional. I took the liberty of bringing the same AltClick functionality to internals tanks as well, allowing you to toggle your internals on quickly!

In my opinion, my changes make the game a bit more realistic and fun. I think that my refactor also definitely makes the code more maintainable. The refactor decouples some shared logic between air tank functions and human/carbon functions, separating those concerns more appropriately and making the system more bug-resistant in the future.

## Changelog

:cl: A.C.M.O.
add: Hermetically sealed and space-worthy helmets can now be used as breathing apparatuses!
add: Internals air tanks can now be toggled via AltClick.
fix: Breath masks are fixed, and can now be toggled via AltClick.
refactor: Refactored and deduplicated a significant portion of internal air tank functions.
fix: Fixed a bug that caused any type of equipped mask to be auto-adjusted when toggling internals.
fix: Fixed a bug that prevented toggling of internals via the stripping menu if the mob only had a breathing tube.
fix: Fixed a bug that allowed two mobs to breathe from the same air tank after exchanging it between them.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -339,12 +339,14 @@
...
 
-//Fourth and final link in a breath chain
+/// Fourth and final link in a breath chain
 /mob/living/carbon/proc/handle_breath_temperature(datum/gas_mixture/breath)
 	// The air you breathe out should match your body temperature
 	breath.temperature = bodytemperature
 
+/// Attempts to take a breath from the external or internal air tank.
 /mob/living/carbon/proc/get_breath_from_internal(volume_needed)
+<<<<<<< HEAD
```
Comment (Floofies) [inline code/modules/mob/living/carbon/life.dm:349]: Delete this old segment completely

File: code/modules/mob/living/carbon/life.dm [hunk 2]
```diff
@@ -339,12 +339,14 @@
...
 	if(internal)
 		if(internal.loc != src && !(wear_mask.clothing_flags & MASK_EXTEND_RANGE)) //SKYRAT EDIT ANESTHETIC MACHINE. ORIGNIAL CODE: if(internal.loc != src)
```
Comment (Floofies) [inline code/modules/mob/living/carbon/life.dm:351]: I see the edit here looks for one more clothing flag. We can delete this section and that can be added to the checks in `invalid_internals()`

**Comment by Floofies** [comment]
This conflicts with a couple things from PR #14452 so I am making a PR for this PR to make it merge-able!

This PR adds "externals" which allows the anesthetic machine to work despite not being equipped by the mob, removing the need for extra conditionals which check for extra flags. The anesthetic machine directly sets the internals of the mob when it should instead use `carbon.open_internals(tank, TRUE)` and `carbon.close_externals()`, notably here, on lines 102 and 131: https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/modules/medical/code/anesthetic_machine.dm#L102

**Comment by Floofies** [comment]
Created PR #17296 to fix this PR before it is merged with master.

File: code/game/objects/items/tanks/tank_types.dm [hunk 3]
```diff
@@ -8,6 +8,16 @@
  * Generic
  */
 
+/// Allows carbon to toggle internals via AltClick of the equipped tank.
+/obj/item/tank/internals/AltClick(mob/user)
+	..()
```
Comment (GoldenAlpharex) [inline code/game/objects/items/tanks/tank_types.dm:13]: I feel like that's going to be a problem at some point, but whatever.

**Review by GoldenAlpharex — APPROVED** [review]
Should be good to go now!

**Pull request merged**